src/components/data/test/DataTable-test.js

```diff
@@ -74,7 +74,7 @@ describe('DataTable', () => {
     );
   }
 
-  const createComponent = (Component, props = {}) => {
+  const createComponent = ({ Component = DataTable, ...props } = {}) => {
```

Comment:
Not directly related with the main changes in this PR, but to avoid duplicating `createComponent(DataTable)` over and over, I added a new `Component` pseudo-prop/option here, which defaults to `DataTable`.

With this, we can just call `createComponent()` or `createComponent(props)` in most of the cases, and when a different component is needed, we can additionally pass `createComponent({ Component: Whatever, ...otherProps })`.